docs/sql/statements/refresh.rst

```diff
@@ -22,30 +22,80 @@ Synopsis
 Description
 ===========
 
-The REFRESH TABLE command refreshes one or more tables and all its rows, making
-all changes made to that table available to all further commands.
+The REFRESH TABLE command refreshes one or more tables, making all changes made
+to that table visible to subsequent commands.
 
-The ``PARTITION`` clause can be used to only refresh specific partitions of a
-partitioned table. All columns by which a table is partitioned are required.
+The ``PARTITION`` clause can be used to refresh specific partitions of a
+partitioned table instead of all partitions. The ``PARTITION`` clause requires
+all partitioned columns with the values that identify the partition as
+argument.
 
 In case the ``PARTITION`` clause is omitted all open partitions will be
 refreshed. Closed partitions are not refreshed.
 
-For performance reasons refreshing all open partitions should be avoided if
-possible.
-
 See :ref:`partitioned_tables` for more information on partitioned tables.
 
-Without issuing this statement there is no guarantee that a change (``UPDATE``,
-``DELETE``, ``COPY`` or ``INSERT``) will be seen by an immediately following
-``SELECT`` statement for the row(s) affected by this change. By default a
-refresh is scheduled periodically.
 
-Internally a refresh opens a new table reader, which is then used by all
-subsequent searches on the affected table.
+For performance reasons, refreshing tables or partitions should be avoided as
+it is an expensive operation. By default CrateDB periodically refreshes the
+tables anyway. See :ref:`refresh_data` and :ref:`sql_ref_refresh_interval` for
+more information about the periodic refreshes.
+
+Without an explicit ``REFRESH``, other statements like ``UPDATE``, ``DELETE``
+or ``SELECT`` won't see data until the periodic refresh happens.
+
+An exception to that are statements which can filter on a primary key with an
+exact match on one or more values. For example, looking up a single document in
```

Comment:
if i got this right all primary key values have to be addressed, right?
```suggestion
exact match on all primary key values. For example, looking up a single document in
```

Reply:
It was a bit amgigious. In the context of a single record all primary key values must be present.
What I wanted to say is that you can lookup multiple records - but I think the later example shows that anyhow, so I used your suggestion and tweaked it a bit further